### PR TITLE
sspBC: Adapter update

### PR DESCRIPTION
**Pull request by mabielinski-wpm**
This PR updates the adapter and moves operations to our backend service.

### CODE REVIEW

**Comment by mabielinski-wpm** [comment]
Hello @bsardo,
it's been a while since I opened this pull request and I am wondering if I should keep waiting for a review or is there anything else I should do on my side? Thanks!

File: static/bidder-info/sspBC.yaml [hunk 1]
```diff
@@ -11,3 +11,6 @@ userSync:
   iframe:
     url: https://ssp.wp.pl/bidder/usersync?tcf=2&redirect={{.RedirectURL}}
     userMacro: $UID
+  redirect:
+    url: https://ssp.wp.pl/v1/sync/prebid-server/pixel?gdpr={{.GDPR}}&gdpr_consent={{.GDPRConsent}}&redirect={{.RedirectURL}}
```
Comment (pm-isha-bharti) [inline static/bidder-info/sspBC.yaml:15]: Verifed, executes with 302 redirect and` /setuid` is called with `uids` cookie set has `adipolo` bidder

Comment (mabielinski-wpm) [reply static/bidder-info/sspBC.yaml:15]: Hello @pm-isha-bharti,
just wanted to ask - have you finished the review, or are you planning to continue it?
Thanks!

File: static/bidder-info/sspBC.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-endpoint: "https://ssp.wp.pl/v1/bidder/prebidserver"
+endpoint: "https://ssp.wp.pl/v2/bidder/prebidserver"
```
Comment (pm-isha-bharti) [inline static/bidder-info/sspBC.yaml:1]: Endpoint is reachable
```
 curl -i --location --request POST https://ssp.wp.pl/v2/bidder/prebidserver
HTTP/2 204
server: nginx
date: Thu, 03 Jul 2025 08:56:37 GMT
accept-ch: device-memory, dpr, width, viewport-width, rtt, downlink, ect
accept-ch-lifetime: 604800
access-control-allow-credentials: true
access-control-allow-origin: *
vary: Origin
```

**Comment by mabielinski-wpm** [comment]
Hello @pm-isha-bharti, could you please finish the code review? Thanks!

File: adapters/sspBC/sspbc.go [hunk 3]
```diff
@@ -1,135 +1,70 @@
...
-	}
-
-	requestJSON, err := json.Marshal(formattedRequest)
-	if err != nil {
-		return nil, []error{err}
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, extraRequestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	sspBcRequest := &requestData{
+		Request: request,
+		RequestInfo: &requestInfo{
+			PbsEntryPoint: extraRequestInfo.PbsEntryPoint,
```
Comment (pm-isha-bharti) [inline adapters/sspBC/sspbc.go:54]: Can you add JSON test coverage for this line where `PbsEntryPoint` is not empty?

File: adapters/sspBC/sspbc.go [hunk 4]
```diff
@@ -157,253 +92,49 @@ func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest
...
-		request.Site.Domain = siteURL.Hostname()
-	}
-
-	// set TEST Flag
-	if requestType == requestTypeTest {
-		request.Test = 1
+func buildAdapterEndpoint(endpoint string, adapterVersion string) (string, error) {
+	endpointURL, err := url.Parse(endpoint)
+	if err != nil {
+		return "", fmt.Errorf("unable to parse endpoint URL: %w", err)
```
Comment (pm-isha-bharti) [inline adapters/sspBC/sspbc.go:132]: Can you add JSON test coverage for this error scenario?

**Comment by pm-isha-bharti** [comment]
Added comments for code coverage, rest looks good to me

**Comment by przemkaczmarek** [comment]
@mabielinski-wpm Hi, LGTM, but please add tests that cover video, native, and audio as well.

File: adapters/sspBC/sspbc.go (hunk 3 above)
Comment (mabielinski-wpm) [reply adapters/sspBC/sspbc.go:54]: Added (video.json file)

File: adapters/sspBC/sspbc.go (hunk 4 above)
Comment (mabielinski-wpm) [reply adapters/sspBC/sspbc.go:132]: Added

**Comment by mabielinski-wpm** [comment]
> @mabielinski-wpm Hi, LGTM, but please add tests that cover video, native, and audio as well.

Hello, added, thanks for the review!

**Review by przemkaczmarek — DISMISSED** [review]

**Comment by mabielinski-wpm** [comment]
> Added comments for code coverage, rest looks good to me

Hello @pm-isha-bharti, I've added code coverage, could you please take a look?

**Comment by mabielinski-wpm** [comment]
Hi @pm-isha-bharti, I wanted to remind you to finish the review.

File: adapters/sspBC/sspbctest/video/video.json [hunk 5]
```diff
@@ -0,0 +1,133 @@
+{
```
Comment (pm-isha-bharti) [inline adapters/sspBC/sspbctest/video/video.json]: Can we move `video.json` to the `exemplary` folder?

Comment (mabielinski-wpm) [reply adapters/sspBC/sspbctest/video/video.json]: @pm-isha-bharti moved
